### PR TITLE
tools:scripts: Added build support for Maxim

### PR DESCRIPTION
**Pull request by CiprianRegus**
Added maxim.mk file and a script to configure VSCode for building
and debugging on Maxim platform.

Signed-off-by: Ciprian Regus <ciprian.regus@analog.com>

### CODE REVIEW

**Comment by CiprianRegus** [comment]
V2:
- SDK source and header files are now automatically linked.
- Ported the ADXL355 to Maxim platform for testing the build system.

File: projects/eval-adxl355-pmdz/src/main.c [hunk 1]
```diff
@@ -7,38 +7,77 @@
...
 	struct adxl355_dev *adxl355;
-
+	struct spi_platform_ops *spi_ops;
 	int ret;
-	stm32_init();
 
+#ifdef STM32_PLATFORM
+	stm32_init();
+	HAL_Init();
+	SystemClock_Config();
```
Comment (rbolboac) [inline projects/eval-adxl355-pmdz/src/main.c:54]: These lines are not needed anymore. stm32_init is performing these actions instead.

File: projects/eval-adxl355-pmdz/src/main.c [hunk 2]
```diff
@@ -7,38 +7,77 @@
...
+#include "maxim_uart.h"
+#include "maxim_stdio.h"
+#endif
+
 #include "adxl355.h"
 
-#define SPI_DEVICE_ID	 1
+#ifdef STM32_PLATFORM
+#define UART_ID			5
+#define SPI_DEVICE_ID	 	1
```
Comment (rbolboac) [inline projects/eval-adxl355-pmdz/src/main.c:29]: nip: check if indentation is correct

File: projects/eval-adxl355-pmdz/src/main.c [hunk 3]
```diff
@@ -7,38 +7,77 @@
...
 #define SPI_CS			15
 #define SPI_CS_PORT		GPIOA
+#endif
+
+#ifdef MAXIM_PLATFORM
+#define UART_ID			1
+#define SPI_DEVICE_ID		0
```
Comment (rbolboac) [inline projects/eval-adxl355-pmdz/src/main.c:36]: nip: check if indentation is correct

**Comment by CiprianRegus** [comment]
V3:
- Deleted extra tabs in defines.
- Removed HAL_Init() and SystemClock_Config().
- Added path to svd file to debug configuration.
- Running sdkopen updates the current configurations with the TARGET value.

**Review by adrimbarean — COMMENTED** [review]
One question and a note. Upon running the build on a windows machine the linker threw the `[object] uses VFP register arguments, [elf] does not` error. This error might be due to my misunderstanding on how to run the build properly so I would suggest having some recepy for running the build written down somewhere. Just `make` didn't work. The command I used was `make PLATFORM=maxim MAXIM_LIBRARIES=C:\MaximSDK\Libraries`

**Review by buha — APPROVED** [review]

**Pull request merged**